contrib/net/http/trace.go

```diff
@@ -94,3 +95,15 @@ func (w *responseWriter) WriteHeader(status int) {
 	w.ResponseWriter.WriteHeader(status)
 	w.status = status
 }
+
+// HeaderTagsFromRequest matches req headers to user-defined list of header tags
+// and creates span tags based on the header tag target and the req header value
+func HeaderTagsFromRequest(req *http.Request, headersAsTags map[string]string) ddtrace.StartSpanOption {
+	return func(cfg *ddtrace.StartSpanConfig) {
+		for k := range req.Header {
+			if ht, ok := headersAsTags[strings.ToLower(k)]; ok && !strings.HasPrefix(strings.ToLower(k), "x-datadog-") {
+				cfg.Tags[ht] = strings.Join(req.Header.Values(k), ",")
+			}
+		}
+	}
+}
```

Comment:
Why have this function here, instead of in `gopkg.in/DataDog/dd-trace-go.v1/contrib/internal/httptrace`?
I ask because the package description for internal/httptrace is
> // Package httptrace provides functionalities to trace HTTP requests that are commonly required and used across contrib/** integrations.

This would also eliminate the need to import both internal/httptrace and contrib/net/http within all of those contrib packages, which is making the import aliasing a bit messy.

Reply:
Same question with `ConvertHeaderToTag`. Notice that none of the other functions in `ddtrace/tracer/util.go` are exported. For `ConvertHeaderToTag`, it definitely seems like having the code in an internal shared utility package makes more sense. Otherwise, we're exposing that function as part of the public API, which won't have any use for consumers of our package.

Reply:
Note for myself as I look into this:

- originally chose to put HeaderTagsFromRequest in here b/c I though there were some import issues importing httptrace into the net/http pkg, but let me revisit this
- ConvertHeaderToTag is used in the global tracer package b/c the global config also needs access to it, hence why I put it in tracer/util rather than a contrib pkg, but will revisit this

Reply:
Additional notes:

- no circular import issues, actually. However, mux integration currently imports the net/http pkg with import name "httptrace", so we will have to change some import names across the board anyway (i.e, either change net/http import names away from httptrace so that the httptrace pkg can use it, or rename httptrace pkg/ rename httptrace pkg imports to something else e.g, httptraceinternal
-